model/infinispan/src/main/java/org/keycloak/connections/infinispan/DefaultInfinispanConnectionProviderFactory.java

```diff
@@ -477,16 +477,16 @@ private void registerSystemWideListeners(KeycloakSession session) {
         ClusterProvider cluster = session.getProvider(ClusterProvider.class);
         cluster.registerListener(REALM_CLEAR_CACHE_EVENTS, (ClusterEvent event) -> {
             if (event instanceof ClearCacheEvent) {
-                sessionFactory.invalidate(ObjectType._ALL_);
+                sessionFactory.invalidate(session, ObjectType._ALL_);
```

Comment:
The short-lived `session` would not be available once `REALM_CLEAR_CACHE_EVENTS` is received. This applies to the rest as well
```suggestion
                sessionFactory.invalidate(null, ObjectType._ALL_);
```